Applications/spire/source/book_view_ui_tester/random_book_view_model.cpp

```diff
@@ -44,6 +44,41 @@ void RandomBookViewModel::set_period(time_duration period) {
   }
 }
 
+void RandomBookViewModel::publish(const BookQuote& quote) {
+  if(quote.m_quote.m_side == Side::BID) {
```

Comment:
Refactor this entire section. Use the function Nexus::GetDirection(Side s) to unify the handling of bids and asks.

For example instead of:

```
if(side == BID) {
  if(price1 > price2) {
    ...
  }
} else {
  if(price1 < price2) {
   ...
  }
}
```

Unify that by doing the following:

```
auto direction = GetDirection(side);
if(direction * price1 > direction * price2) {
  ...
}
```